module_names:
  - "map"
  - "sensing"
  - "perception"
  - "localization"
  - "planning"
  - "control"
  - "vehicle"
  - "system"

# Topic Config
## module: string
## name: string
## timeout[s]: double (0: none)
## warn_rate[hz]: double (0: none)
topic_configs:
  - module: "map"
    name: "/map/vector_map"
    timeout: 0.0
    warn_rate: 0.0

  - module: "map"
    name: "/map/pointcloud_map"
    timeout: 0.0
    warn_rate: 0.0

  - module: "perception"
    name: "/perception/object_recognition/objects"
    timeout: 1.0
    warn_rate: 5.0

  - module: "localization"
    name: "/initialpose"
    timeout: 0.0
    warn_rate: 0.0

  - module: "planning"
    name: "/planning/mission_planning/route"
    timeout: 0.0
    warn_rate: 0.0

  - module: "planning"
    name: "/planning/scenario_planning/trajectory"
    timeout: 1.0
    warn_rate: 5.0

  - module: "control"
    name: "/control/trajectory_follower/control_cmd"
    timeout: 1.0
    warn_rate: 5.0

  - module: "control"
    name: "/control/vehicle_cmd"
    timeout: 1.0
    warn_rate: 5.0

  - module: "vehicle"
    name: "/vehicle/status/twist"
    timeout: 1.0
    warn_rate: 5.0

  - module: "vehicle"
    name: "/vehicle/status/steering"
    timeout: 1.0
    warn_rate: 5.0

  - module: "system"
    name: "/system/emergency/control_cmd"
    timeout: 1.0
    warn_rate: 5.0

# Param Config
## name: string
param_configs:
  - module: "vehicle"
    name: "/vehicle_info"

# TF Config
## from: string
## to: string
tf_configs:
  - module: "localization"
    from: "map"
    to: "base_link"
    timeout: 1.0
